doc/src/manual/types.md

```diff
@@ -1,66 +1,58 @@
-detail in [Methods](@ref), but is rooted in the type system presented here.
-
-The default behavior in Julia when types are omitted is to allow values to be of any type. Thus,
-one can write many useful Julia functions without ever explicitly using types. When additional
-expressiveness is needed, however, it is easy to gradually introduce explicit type annotations
-into previously "untyped" code. Adding annotations serves three primary purposes: to take advantage
-of Julia's powerful multiple-dispatch mechanism,  to improve human readability, and to catch
-programmer errors.
-
-Describing Julia in the lingo of [type systems](https://en.wikipedia.org/wiki/Type_system), it
-is: dynamic, nominative and parametric. Generic types can be parameterized, and the hierarchical
-relationships between types are [explicitly declared](https://en.wikipedia.org/wiki/Nominal_type_system),
-rather than [implied by compatible structure](https://en.wikipedia.org/wiki/Structural_type_system).
-One particularly distinctive feature of Julia's type system is that concrete types may not subtype
-each other: all concrete types are final and may only have abstract types as their supertypes.
+Types in Julia establish distinctions between different kinds of data,
+and are used by the compiler to infer the intended use of those data.
+Programming languages have traditionally employed one of two quite different type systems:
+static type systems, where expressions must have a computable type before the execution of the program,
+and dynamic type systems, where types are only computed at run time,
+when the actual values manipulated by the program are available.
+Statically typed languages typically offer faster execution of programs,
+while treating any data of unknown or invalid type as an error.
+In these languages, the programmer is responsible for ensuring the validity of all types of data that the program may encounter.
+Dynamically typed languages, on the other hand, do not explicitly check types,
+and types of data only become invalid when they fail to support run-time operations.
+
+Julia's type system is dynamic, but gains some of the advantages of static type systems
+by allowing optional type annotations. Type annotations are important for defining efficient new data structures,
+can help enforce correctness and clarity, and can sometimes improve efficiency
+(although in many contexts the Julia compiler infers types automatically.
+Type annotations also play a central role in Julia because they control method dispatch:
+functions can be extended to new behaviours for different argument types.
+Method dispatch is explored in detail in the [Methods](@ref) section, but is rooted in the type system presented here.
+
+All values in a Julia program belong to exactly one concrete type.
+All types belong to a single type tree, i.e. they are all at least related to the [`Any`](@ref) type.
+User-defined types are treated the same as built-in types.
```

Comment:
```suggestion
```

either this is wrong, or it needs more explanation about what "the same" means here. because there are certainly built-in types which get very special treatment